config/checkstyle/checkstyle-suppressions.xml

```diff
@@ -5,4 +5,6 @@
 <suppressions>
 	<suppress files="[\\/]src[\\/]test[\\/]java[\\/]" checks="JavadocVariable" />
 	<suppress files="[\\/]src[\\/]test[\\/]java[\\/]" checks="JavadocMethod" />
+	<suppress files="[\\/]src[\\/]test[\\/]java[\\/]" checks="JavadocMethod" />
+	<suppress files="RestDocsSnippetBlockMacro\.java" checks="RedundantModifier" />
```

Comment:
Asciidoctor needs to instantiate the macro using reflection(?). If I reduce the visibility of the constructor I get `org.jruby.exceptions.RaiseException: (TypeError) asciidoctor: FAILED: <stdin>: Failed to parse source, no public constructors for Java::OrgSpringframeworkRestdocsAsciidoctorExtensions::RestDocsSnippetBlockMacro`. I don't see any possibility to use another registration method and also provide the macro's block name.